pandas/core/ops.py

```diff
@@ -663,6 +663,10 @@ def wrapper(left, right, name=name, na_op=na_op):
                     not isinstance(lvalues, pd.DatetimeIndex)):
                 lvalues = lvalues.values
 
+            if (hasattr(rvalues, 'values') and
+                    not isinstance(rvalues, pd.DatetimeIndex)):
```

Comment:
IIRC this *should* be done inside ``_TimeOp`` actually; its should return ndarrays. The code is pretty tricky / complicated, but it should go there. If you want to have a try at it. I dont know when/where the lines above were added (they should not be there). 

Having code to do things in multiple places is something I really really try not to do. It does happen, but is just totally confusing.